core/src/main/java/jenkins/install/InstallUtil.java

```diff
@@ -77,6 +85,10 @@ public static InstallState getInstallState() {
         // Neither the top level config or the lastExecVersionFile have a version
         // stored in them, which means it's a new install.
         if (lastRunVersion.compareTo(NEW_INSTALL_VERSION) == 0) {
+            // Edge case: used jenkins 1 but not saved the system config page, the version is not persisted, returns 1.0
+            if(!Jenkins.getInstance().getJobNames().isEmpty()) { // some jobs configured
```

Comment:
:bug: this is not reliable. I presume you are attempting to somehow detect the case that a user “did stuff” (meaning, created items) in Jenkins 1.x but somehow managed to avoid ever saving `/configure`—possible, if very unlikely. But this check will be false if the user created top-level _folders_ but no top-level _jobs_ (or subsequently moved all jobs into folders). `getTopLevelItemNames` would be more appropriate. Or more directly:

```java
if (!Jenkins.getInstance().getItemMap().isEmpty()) { // some items configured
```

(Note that using `getAllItems()` etc. is not only slightly slower, but vulnerable to inaccurate results in case the caller is running with non-`ACL.SYSTEM` authentication.)